src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -2,56 +2,45 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.IO.Pipelines;
 using System.Net.Http;
 using System.Threading.Tasks;
+using System.Threading.Tasks.Channels;
+using Microsoft.AspNetCore.Sockets.Internal;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Sockets.Client
 {
-    public class Connection : IPipelineConnection
+    public class Connection : IChannelConnection<Message>
```

Comment:
I have mixed feelings about Connection being IChannelConnection
- exposing Message
- stopping the connection is ugly and has loose ends (especially you need to drain the channel in order to make the Completion task complete)

I updated issue https://github.com/aspnet/SignalR/issues/51 but we need more to do more thinking here (or maybe it is fine at the low level?)